vscode-client/snippets/snippets.json

```diff
@@ -0,0 +1,214 @@
+        ]
+    },
+    ":-": {
+        "description": "[:-] expansion",
+        "prefix": ":-",
+        "body": [
+            "\"\\${${1:variable}:-${2:default}}\""
+        ]
+    },
+    ":=": {
+        "description": "[:=] expansion",
+        "prefix": ":=",
+        "body": [
+            "\"\\${${1:variable}:=${2:default}}\""
+        ]
+    },
+    ":?": {
+        "description": "[:?] expansion",
+        "prefix": ":?",
+        "body": [
+            "\"\\${${1:variable}:?${2:error_message}}\""
+        ]
+    },
+    ":+": {
+        "description": "[:+] expansion",
+        "prefix": ":+",
+        "body": [
+            "\"\\${${1:variable}:+${2:alternative}}\""
+        ]
+    },
+    "#": {
+        "description": "[#] expansion",
+        "prefix": "#",
+        "body": [
+            "\"\\${${1:variable}#${2:pattern}}\""
+        ]
+    },
+    "##": {
+        "description": "[##] expansion",
+        "prefix": "##",
+        "body": [
+            "\"\\${${1:variable}##${2:pattern}}\""
+        ]
+    },
+    "%": {
+        "description": "[%] expansion",
+        "prefix": "%",
+        "body": [
+            "\"\\${${1:variable}%${2:pattern}}\""
+        ]
+    },
+    "%%": {
+        "description": "[%%] expansion",
+        "prefix": "%%",
+        "body": [
+            "\"\\${${1:variable}%%${2:pattern}}\""
+        ]
+    },
+    "..": {
+        "description": "brace expansion",
+        "prefix": "..",
+        "body": [
+            "{${1:from}..${2:to}}"
+        ]
+    },
+    "echo": {
+        "description": "[echo]",
+        "prefix": "echo",
+        "body": [
+            "echo \"${1:message}\""
+        ]
+    },
+    "printf": {
+        "description": "[printf]",
+        "prefix": "printf",
+        "body": [
+            "printf '%s' \"${1:message}\""
+        ]
+    },
+    "source": {
+        "description": "[source]",
+        "prefix": "source",
+        "body": [
+            "source \"${1:path/to/file}\""
+        ]
+    },
+    "alias": {
+        "description": "[alias]",
+        "prefix": "alias",
+        "body": [
+            "alias ${1:name}=${2:value}"
+        ]
+    },
+    "cd": {
+        "description": "[cd]",
+        "prefix": "cd",
+        "body": [
+            "cd \"${1:path/to/directory}\" || echo \"${2:error_message}\" >&2"
+        ]
+    },
+    "getopts": {
+        "description": "[getopts]",
+        "prefix": "getopts",
+        "body": [
+            "getopts ${1:optstring} ${2:name}"
+        ]
+    },
+    "jobs": {
+        "description": "[jobs]",
+        "prefix": "jobs",
+        "body": [
+            "jobs -x ${1:command}"
+        ]
+    },
+    "kill": {
+        "description": "[kill]",
+        "prefix": "kill",
+        "body": [
+            "kill ${1|-l,-L|}"
+        ]
+    },
+    "let": {
+        "description": "[let]",
+        "prefix": "let",
+        "body": [
+            "let ${1:argument}"
+        ]
+    },
+    "test": {
```

Comment:
`test` seems really useful with all the different options here.

Reply:
`=~` has been added.